doc/nrf/ug_nfc.rst

```diff
@@ -0,0 +1,282 @@
+   Note, however, that chunking is not supported by this library.
+SR (Short Record):
+   Used to determine the size of the payload length field.
+   If the flag is set, the Payload Length occupies 1 byte; otherwise it is 4 bytes long.
+   Note that the NDEF generator supports a Payload Length of 4 bytes only at the moment.
+IL (ID Length present):
+   Indicates whether an ID Length field is present in the header.
+   If the flag is set, the ID Length field is present.
+TNF (Type Name Format):
+   Specifies the structure of the Payload Type field and how to interpret it.
+   The following values are allowed (square brackets contain documentation reference related to the specific type):
+
+   +-------+------------------------------------+
+   | Value | Type Name Format                   |
+   +=======+====================================+
+   | 0x00  | Empty                              |
+   +-------+------------------------------------+
+   | 0x01  | NFC Forum well-known type [NFC RTD]|
+   +-------+------------------------------------+
+   | 0x02  | Media-type [RFC 2046]              |
+   +-------+------------------------------------+
+   | 0x03  | Absolute URI [RFC 3986]            |
+   +-------+------------------------------------+
+   | 0x04  | NFC Forum external type [NFC RTD]  |
+   +-------+------------------------------------+
+   | 0x05  | Unknown                            |
+   +-------+------------------------------------+
+   | 0x06  | Unchanged                          |
+   +-------+------------------------------------+
+   | 0x07  | Reserved                           |
+   +-------+------------------------------------+
+
+.. _ug_nfc_ndef_format_payload:
+
+Record payload
+==============
+
+The content of the payload is application-specific and related to the type of the record.
+For example, in URI records, the payload contains a web address of the page that the polling device should open.
+
+Note that the payload of an NDEF record can contain a nested NDEF message.
+This nested message must be a full NDEF message, consisting of one or multiple NDEF records with the appropriate setting of MB and ME flags.
+
+
+.. _ug_nfc_tnep:
+
+Tag NDEF Exchange Protocol (TNEP)
+*********************************
+
+The Tag NDEF Exchange Protocol (TNEP) is an application-level protocol for sending or retrieving application data units between a polling device and an NFC tag.
+The data units are NFC Data Exchange Format (NDEF) messages.
+
+The protocol operates between the NDEF application layer and the tag.
+The tag must be a device that supports at least one communication protocol for NDEF.
+The polling device must be in poll mode, which means that it sends commands and receives responses.
+
+The NFC tag supports all tag types from Type 2 Tag to Type 5 Tag.
+The polling device communicates with the type of tag corresponding to the tag type. (??? this contradicts the previous sentence that the tag can support whatever ???)
```

Comment:
Is the previous sentence supposed to say that the NFC tag can support any of the tag types?